spanner/session.go

```diff
@@ -924,6 +1004,27 @@ var errGetSessionTimeout = spannerErrorf(codes.Canceled, "timeout / context canc
 // sessions being checked out of the pool.
 func (p *sessionPool) newSessionHandle(s *session) (sh *sessionHandle) {
 	sh = &sessionHandle{session: s, checkoutTime: time.Now(), lastUseTime: time.Now()}
+	if s.isMultiplexed {
+		p.mu.Lock()
+		p.sc.mu.Lock()
+		clientOpt := option.WithGRPCConn(p.sc.connPool.Conn())
+		clientInfo := []string{"gccl", internal.Version}
+		if p.sc.userAgent != "" {
+			agentWithVersion := strings.SplitN(p.sc.userAgent, "/", 2)
+			if len(agentWithVersion) == 2 {
+				clientInfo = append(clientInfo, agentWithVersion[0], agentWithVersion[1])
+			}
+		}
+		p.sc.mu.Unlock()
+		p.mu.Unlock()
+		client, err := vkit.NewClient(context.Background(), clientOpt)
+		if err != nil {
+			return nil
+		}
+		client.SetGoogleClientInfo(clientInfo...)
+		client.CallOptions = s.client.CallOptions
+		sh.client = client
```

Comment:
I wonder whether this is also something that we could do only once when the session pool is created, as the number of connections is bounded (and small), instead of repeating it every time we create a session handle.